pmd-java/src/main/resources/category/java/design.xml

```diff
@@ -1305,9 +1304,11 @@ or
             <property name="xpath">
                 <value>
 <![CDATA[
-//ConditionalExpression[not(PrimaryExpression/*/Literal) and (Expression/PrimaryExpression/*/Literal/BooleanLiteral)]
-|
-//ConditionalExpression[not(Expression/PrimaryExpression/*/Literal) and (PrimaryExpression/*/Literal/BooleanLiteral)]
+//ConditionalExpression[
+((PrimaryExpression/*/Literal/BooleanLiteral) or (Expression/PrimaryExpression/*/Literal/BooleanLiteral))
+and
+not((PrimaryExpression/*/Literal/NullLiteral) or (Expression/PrimaryExpression/*/Literal/NullLiteral))
+]
```

Comment:
Would be nicer if it was 
```suggestion
//ConditionalExpression
  [(Expression|.)/PrimaryExpression/*/Literal/BooleanLiteral]
  [not((Expression|.)/PrimaryExpression/*/Literal/NullLiteral)]
```

Both those expressions match `true ? e1 : e2`, which should be tested now.